Files/App.xaml.cs

```diff
@@ -394,6 +388,64 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
             deferral.Complete();
         }
 
+        // Occurs when an exception is not handled on the UI thread.
+        private static void OnUnhandledException(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
+        {
+            Logger.Error(e.Exception, e.Message);
+
+            var diagnosticInfo = new Dictionary<string, string>()
+            {
+                { "Message", e.Message },
+                { "Exception", e.Exception?.ToString() },
+                { "Culture", SystemInformation.Culture.EnglishName },
+                { "AvailableMemory", SystemInformation.AvailableMemory.ToString("F0") },
+                { "FirstUseTimeUTC", SystemInformation.FirstUseTime.ToUniversalTime().ToString("MM/dd/yyyy HH:mm:ss") },
+                { "OSArchitecture", SystemInformation.OperatingSystemArchitecture.ToString() },
+                { "OSVersion", SystemInformation.OperatingSystemVersion.ToString() }
+            };
+
+            var attachment = ErrorAttachmentLog.AttachmentWithText(
+                $"Exception: {e.Exception}, " +
+                $"Message: {e.Message}, " +
+                $"InnerException: {e.Exception?.InnerException}, " +
+                $"InnerExceptionMessage: {e.Exception?.InnerException?.Message}",
+                "UnhandledException");
+
+            Analytics.TrackEvent("OnUnhandledException", diagnosticInfo);
+            Crashes.TrackError(e.Exception, diagnosticInfo, attachment);
+
+            // suppress and handle it manually.
+            e.Handled = true;
```

Comment:
Don't suppress crash. We should fix app crash by fixing bugs instead of handling all possible exceptions. This PR should add only extended crash log info.

Reply:
Right. If we handled all the exception it will make it harder to find bugs. Let it crash so that we can know that there's an issue without seeing logs.

Reply:
I am making changes so that app will crash only in debug version and not in release version. Is that acceptable??

Reply:
No, it is not acceptable. If smth wrong, app should crash even in release mode.

Reply:
@soumyamahunt If something is wrong, we would like to get the extended crash info logged, however, we should still let the app crash so that we get the report and crash error rate info in AppCenter and the Partner Center dashboard.

Reply:
@yaichenbaum I am using `AppCenter.Crashes.CrashError` to upload error report to AppCenter dashboard. Also, I am using `AppCenter.Analytics.TrackEvent` too. App doesn't need to crash to get error rate info in AppCenter and the Partner Center dashboard.

Reply:
@soumyamahunt Tracking the error rate is very important in order to keep track of the amount of errors across versions. Please address the changes @tsvietOK requested 🙂.

Reply:
> @soumyamahunt Tracking the error rate is very important in order to keep track of the amount of errors across versions. Please address the changes @tsvietOK requested 🙂.

Addressed those changes.